cli/dcoscli/metrics.py

```diff
@@ -258,8 +263,18 @@ def print_task_metrics(url, app_url, summary, json_):
     :rtype: int
     """
 
-    datapoints = _fetch_metrics_datapoints(url) + _fetch_metrics_datapoints(
-        app_url)
+    container_datapoints = []
+    app_datapoints = []
+
+    # In the case of an executor, app data may exist when
+    # container data does not.
+    try:
+        container_datapoints = _fetch_metrics_datapoints(url)
+    except EmptyMetricsException:
+        pass
+
+    app_datapoints = _fetch_metrics_datapoints(app_url)
```

Comment:
Is it possible that we have the other way around, container data exists but not app data?

Reply:
No, there will always be app data because we always pass two app values from the mesos module:

https://github.com/dcos/dcos-metrics/blob/3c96b31b746170981211329d35c96da3eda5843b/mesos_module/container_reader_impl.cpp#L153-L158